s4e-backend/src/main/java/pl/cyfronet/s4e/service/AppUserService.java

```diff
@@ -39,4 +41,15 @@ public AppUser update(AppUser appUser) {
     public Optional<AppUser> findById(Long id) {
         return appUserRepository.findById(id);
     }
+
+    @Transactional
+    public void saveWithGroupUpdate(AppUser appUser) {
+        appUserRepository.save(appUser);
```

Comment:
You don't catch DataIntegrityViolationException here, as in the standard method ::save?